aut_docs/Setup.md

```diff
@@ -9,9 +9,10 @@
 - `cd vms`
 - To run, `python manage.py runserver`. Browse 
   `http://127.0.0.1:8000`
-- To execute tests `python manage.py test`. This will run all unit-tests and
+- Before running tests, make sure to download the latest geckodriver. Unpack and move the binary to `/usr/local/bin`
+- To execute tests `python manage.py test ../tests/`. This will run all unit-tests and
   all functional-tests across all apps. To execute tests of only a particular
-  app, run `python manage.py test <app_name>`
+  app, run `python manage.py test ../tests/<app_name>/`
```

Comment:
I think this is a bit weird approach tbh.

Reply:
Sorry couldn't get your point. :confused:

Reply:
The standard architecture I have seen to be followed in projects which are small but since this is a large project and it will be growing in future too I think its better to have this structure from this point only than to change it on later point when this project become more bigger.
Many large python-based projects use such architecture (separate tests dir which has a folder corresponding to each app).

Reply:
I would strongly recommend against inventing a new standard - specially when you are working with an OSS project like Django (which gets upgraded etc in the meantime). 

See - separate tests directory is what we want - but they should be inside the individual apps. Unit tests should test units and hence - apps.units. 

But I see you are writing E2E tests, which are kind of seperated - true that. In that case, I would recommend this (you need not follow, but I am just guessing loudly): 
1. A separate repository maintaining the E2E tests - as E2E tests should interact only over public interfaces - so no direct DB access is required. 
2. A separate app just for browser tests - like vms.browser_tests/tests/ - this has the advantage of even selectively skipping this if you want.

Reply:
@tonythomas01 updates made according to standards

Reply:
In that case; we'll have two test structure which we have to check at the same time. One for unit-tests and functional tests and another for selenium tests which isn't what we want here as it'd create more trouble than solve some.
I think there are two options for structure then:
1. Individual test dir inside each app (currently present).
2. Common test dir in root (which this PR does).

IMHO its better if admins and maintainers decide over it which suits them better as they are the ones who are looking after the project but I need to start writing tests from this week for gsoc.

Reply:
@Monal5031 I think its a good idea to have the tests under apps and also that manage.py test should run all tests! I thought of using the same method (seperation) you are proposing but that was because I wanted to write the test with pytest. At the end of the day, a lot of complications came up between pytest test runner and django test runner working together  hinge free if sorely pytest ideas and methodologies where enforced. My test are painfully now in accordance with standard django (built on TestCase) I have made less (0 I guess) use of pytest but then it somewhat better becuase pytest can figure it's way arround django but django test runner cant figures its way around pytest structures e.g the conftest file. So both test runners would work now but initially just pytest runner would.

Reply:
Yeah, its better to follow the standard django structure. I have reverted the previous change and kept the tests under apps itself.